Add stm32u5 ospi manager clock in dts

gautierg-st opened the pull request:
The OSPI manager clock was enabled directly using the macro __HAL_RCC_OSPIM_CLK_ENABLE.
This PR fixes it so that this activation is done through the dts.
The different OSPI clocks are now referenced through their (new) names instead of their indexes to make things clearer.

gautierg-st committed:
dts: arm: st: Add clock names for all STM32 OSPI instances in dts

For all STM32 featuring octospi, clock-names are added to use them
instead of indexing for configuring the clock.
For U5 series, a third clock is added for the OSPI manager.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
drivers: flash: Use dts to configure STM32 OSPI manager clock

Instead of calling __HAL_RCC_OSPIM_CLK_ENABLE() to enable the OSPI
manager clock, we now use a new clock binding in the dts.

In order to avoid confusion between the different clocks, the driver
is modified to select the clock based on their names instead of indexes.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

erwango commented on drivers/flash/flash_stm32_ospi.c:
```diff
@@ -1720,33 +1727,39 @@ static int flash_stm32_ospi_init(const struct device *dev)
+		LOG_ERR("Could not enable OSPI Manager clock");
+		return -EIO;
+	}
+#endif
```
Break a line after `#endif`

erwango commented on drivers/flash/flash_stm32_ospi.c:
```diff
@@ -80,10 +80,17 @@ struct stream {
 	OCTOSPI_TypeDef *regs;
-	const struct stm32_pclken *pclken; /* clock subsystem */
-	size_t pclk_len; /* number of clock subsystems */
+	struct stm32_pclken pclken; /* clock subsystem */
```
Make all of these const

erwango commented on drivers/flash/flash_stm32_ospi.c:
```diff
@@ -1962,16 +1974,22 @@ static int flash_stm32_ospi_init(const struct device *dev)
 
 static void flash_stm32_ospi_irq_config_func(const struct device *dev);
 
-#define STM32_OSPI_NODE DT_INST_PARENT(0)
-
 PINCTRL_DT_DEFINE(STM32_OSPI_NODE);
 
-static const struct stm32_pclken pclken_id[] = STM32_DT_CLOCKS(STM32_OSPI_NODE);
+//static const struct stm32_pclken pclken_id[] = STM32_DT_CLOCKS(STM32_OSPI_NODE);
```
Delete this line

erwango commented on drivers/flash/flash_stm32_ospi.c:
```diff
@@ -1962,16 +1974,22 @@ static int flash_stm32_ospi_init(const struct device *dev)
+#if DT_CLOCKS_HAS_NAME(STM32_OSPI_NODE, ospi_mgr)
+	.pclken_mgr = {.bus = DT_CLOCKS_CELL_BY_NAME(STM32_OSPI_NODE, ospi_mgr, bus),
+		       .enr = DT_CLOCKS_CELL_BY_NAME(STM32_OSPI_NODE, ospi_mgr, bits)},
+#endif
```
Since you're using clock-names systematically, you should also set `clock-names` property as `required: true` in the binding. This will help getting meaningful error messages in case it is forgotten.

gautierg-st committed:
dts: bindings: ospi: Require STM32 OSPI clock-names binding

Following the modification of the STM32 OSPI driver, the clock-names
binding is now required

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

FRASTM reviewed (dismissed):
LGTM
With stm32U5 is there any possible error/warning to check that OSPI mgr clock is actually set
